.circleci/config.yml

```diff
@@ -1,122 +1,91 @@
       - run:
-          name: Upload coverage result to codecov
-          command: bash <(curl -s https://codecov.io/bash)
-          # Ref: https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute
-          when: always
+          name: "Upload coverage data to codecov"
+          # Send result to CodeCov
+          command: bash <(wget -qO - https://codecov.io/bash)
       - run:
-          name: Skip image building for pull requests
-          command: if [[ -n "$CIRCLE_PULL_REQUEST" ]] ; then circleci-agent step halt; fi
+          # Install GolangCI-Lint, this might be better off as a separate job for separated out ✓
+          name: "Setup GolangCI-Lint"
+          command: wget https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh -O - | sh -s -- -b $(go env GOPATH)/bin v1.39.0
       - run:
-          name: Build Image
-          command: make build IMGTAG=${IMGTAG}
+          name: "Run GolangCI-Lint"
+          command: golangci-lint run --timeout 5m
+      # TODO: Find if there's a way to get different checks without running the builds in different jobs
       - run:
-          name: Save kubera-auth docker image
-          command: docker save -o /tmp/workspace/${KUBERA_AUTH}.tar ${REPONAME}/${KUBERA_AUTH}:${IMGTAG}
+          name: "Build binary"
+          command: CGO_ENABLED=0 go build -o ./server ./src
       - persist_to_workspace:
-          root: /tmp/workspace
+          root: ~/project
           paths:
-            - kubera-auth.tar
-            - env-vars
-
-  auth-push:
-    machine:
-      image: ubuntu-1604:201903-01
-    working_directory: ~/kubera-auth
+          # NOTE: Only these files will be present in the next CircleCI job
+          - server
+          - templates
+          - Dockerfile
+          - Makefile
+          # Persist ~/project/{server, templates, Dockerfile, Makefile} to workspace, to be used in build job
+  build:
+    executor:
+      name: go/default
+      tag: '1.16.2'
     steps:
+      # NOTE: This job doesn't have the code
       - attach_workspace:
-          at: /tmp/workspace
-      - run: |
-          cat /tmp/workspace/env-vars >> $BASH_ENV
-          source $BASH_ENV
-      - checkout
+          at: ~/project
+      - run: ls -la ~/project
+      # depends on completion of test job
+      - run: docker login -u $DNAME -p $DPASS
+      # https://circleci.com/docs/2.0/building-docker-images/
+      - setup_remote_docker
+      # Build the binary in the go container
       - run:
-          name: Loading kubera auth docker image
+        # The image tagging logic goes here, the env variables aren't exported to further steps
+          name: "Build a docker image"
           command: |
-            docker load -i /tmp/workspace/${KUBERA_AUTH}.tar
-      - run:
-          name: Docker images
-          command: docker images
-      - run:
-          name: Pushing kubera auth
-          command: make push IMGTAG=${IMGTAG} BUILD_TYPE=${BUILD_TYPE}
```

Comment:
What is the $BUILD_TYPE and don't we need that now?

Reply:
Response from @vharsh 
BUILD_TYPE env variable wasn't being used anywhere in Makefile, it was being used in the scripts which are there in hack/pushdirectory.